extensions/data-workspace/src/test/dialogs/newProjectDialog.test.ts

```diff
@@ -48,25 +53,27 @@ suite('New Project Dialog', function (): void {
 		dialog.model.name = `TestProject_${new Date().getTime()}`;
 		dialog.model.location = os.tmpdir();
 		dialog.workspaceInputBox!.value = 'test';
-		should.equal(await dialog.validate(), false, 'Validation should fail because workspace does not end in .code-workspace');
+		should.equal(await dialog.validateNewWorkspace(false), false, 'Validation should fail because workspace does not end in .code-workspace');
```

Comment:
FYI - while looking at this code I noticed that dialogBase is implementing validate to always return true. It seems to me that the better thing would be to have validate be an abstract function instead so you're sure each page is implementing it correctly

Reply:
updated it to be abstract